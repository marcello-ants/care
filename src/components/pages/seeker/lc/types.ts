import { GeneralProviderProfile } from '@/types/seeker';

export interface ProviderProfile extends GeneralProviderProfile {
  numberOfReviews: number;
  qualities?: ProviderQualities;
  services?: ProviderServices;
  firstName: string;
}

export interface ProviderQualities {
  certifiedNursingAssistant?: boolean;
  certifiedRegisterNurse?: boolean;
  certifiedTeacher?: boolean;
  childDevelopmentAssociate?: boolean;
  comfortableWithPets?: boolean;
  cprTrained?: boolean;
  crn?: boolean;
  doesNotSmoke?: boolean;
  doula?: boolean;
  earlyChildDevelopmentCoursework?: boolean;
  earlyChildhoodEducation?: boolean;
  firstAidTraining?: boolean;
  nafccCertified?: boolean;
  ownTransportation?: boolean;
  specialNeedsCare?: boolean;
  trustlineCertifiedCalifornia?: boolean;
}

export interface ProviderServices {
  carpooling?: boolean;
  craftAssistance?: boolean;
  errands?: boolean;
  groceryShopping?: boolean;
  laundryAssistance?: boolean;
  lightHousekeeping?: boolean;
  mealPreparation?: boolean;
  swimmingSupervision?: boolean;
  travel?: boolean;
}
