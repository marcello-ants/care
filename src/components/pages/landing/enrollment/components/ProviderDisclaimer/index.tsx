// External Dependencies
import { Typography, Link } from '@care/react-component-lib';

// Styles
import useStyles from './styles';

export default function ProviderDisclaimer() {
  // Styles
  const classes = useStyles();

  return (
    <Typography careVariant="disclaimer1" className={classes.disclaimer}>
      <span>
        You must be 18 years or older to use Care.com. By clicking &quot;Submit,&quot; you agree to
        our
      </span>{' '}
      <Link href="/terms-of-use-p1012.html" careVariant="disclaimer1">
        Terms of Use
      </Link>{' '}
      <span>and</span>{' '}
      <Link href="/privacy-policy-p1013.html" careVariant="disclaimer1">
        Privacy Policy
      </Link>
      <span>.</span>
    </Typography>
  );
}
