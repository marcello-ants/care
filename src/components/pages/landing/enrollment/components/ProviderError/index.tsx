// External Dependencies
import { Box } from '@material-ui/core';
import { Icon24UtilityWarning } from '@care/react-icons';
import { Typography } from '@care/react-component-lib';

// Styles
import useStyles from './styles';

function ProviderError() {
  // Styles
  const classes = useStyles();

  return (
    <Box className={classes.errorBanner}>
      <Box>
        <Icon24UtilityWarning color="#CF230C" />
      </Box>
      <Box>
        <Typography variant="h4">Whoops! Something went wrong</Typography>
        <Typography careVariant="body3">
          There was an error when we tried to submit your request. Please try again.
        </Typography>
      </Box>
    </Box>
  );
}

export default ProviderError;
