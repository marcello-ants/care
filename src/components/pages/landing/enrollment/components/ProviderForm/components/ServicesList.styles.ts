import { makeStyles } from '@material-ui/core';

const useStyles = makeStyles((theme) => ({
  empty: {
    '& > .MuiSelect-select': {
      color: theme.palette.care?.grey[700],
      fontWeight: 'normal',
    },
  },
  inputContainer: {
    width: '100%',
    '& > div': {
      padding: 0,
    },
  },
  error: {
    color: theme?.palette?.care?.red[800],
    marginLeft: theme.spacing(1.25),
    marginTop: theme.spacing(0.5),
  },
}));

export default useStyles;
