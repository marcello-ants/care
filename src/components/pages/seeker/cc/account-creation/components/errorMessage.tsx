import React from 'react';
import { Box, Grid } from '@material-ui/core';
import { Alert } from '@material-ui/lab';

type ErrorMessageData = {
  errorMessageText: string;
};

const ErrorMessageNotification = (props: ErrorMessageData) => {
  const { errorMessageText } = props;

  return (
    <Grid item xs={12}>
      <Box mb={3}>
        <Alert severity="error">{errorMessageText}</Alert>
      </Box>
    </Grid>
  );
};

export default ErrorMessageNotification;
