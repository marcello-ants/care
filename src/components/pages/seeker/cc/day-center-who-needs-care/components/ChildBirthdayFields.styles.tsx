// External Dependencies
import { makeStyles } from '@material-ui/core';
import { theme } from '@care/material-ui-theme';

const useStyles = makeStyles({
  childContainer: {
    padding: theme.spacing(0),
  },
  childContainerNth: {
    marginTop: theme.spacing(4),
    position: 'relative',
  },
  optionsContainer: {
    display: 'flex',
    flexDirection: 'row',
  },
  fullWidth: {
    width: '100%',
  },
  error: {
    color: theme.palette.care.red[800],
    paddingLeft: theme.spacing(1),
  },
  selectContainer: {
    marginTop: theme.spacing(1),
  },
  monthSelect: {
    marginRight: theme.spacing(0.5),
    '& label': {
      paddingLeft: theme.spacing(1),
    },
  },
  yearSelect: {
    marginLeft: theme.spacing(0.5),
    '& label': {
      paddingLeft: theme.spacing(1),
    },
  },
  removeChildButton: {
    position: 'absolute',
    top: '0',
    right: '0',
    width: 'auto',
    height: 'auto',
    lineHeight: 'unset',
    minWidth: 'auto',
    padding: theme.spacing(0),
    color: theme.palette.care.blue[700],
    '&:hover': {
      backgroundColor: 'transparent',
    },
  },
});

export default useStyles;
