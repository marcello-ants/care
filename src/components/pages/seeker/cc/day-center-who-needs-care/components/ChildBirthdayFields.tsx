// External Dependencies
import React, { useMemo } from 'react';
import { Button, FormHelperText, Grid, InputLabel, MenuItem } from '@material-ui/core';
import { Select, Typography } from '@care/react-component-lib';

// Internal Dependencies
import { ChildDateOfBirth } from '@/types/seekerCC';
import { getBirthAvailableMonthsAndYears } from '../utils';
import useStyles from './ChildBirthdayFields.styles';

// Interfaces
interface TErrorProps {
  year?: string;
  month?: string;
}

type SelectInputProps = React.ComponentProps<typeof Select>;
interface IChildBirthdayFieldsProps {
  child: ChildDateOfBirth;
  onRemoveChild: () => void;
  onChange: SelectInputProps['onChange'];
  childIndex: number;
  error?: TErrorProps;
  allow9MonthsOutFromNow?: boolean;
}

function ChildBirthdayFields({
  child,
  childIndex: i,
  onChange,
  onRemoveChild,
  error,
  allow9MonthsOutFromNow,
}: IChildBirthdayFieldsProps) {
  const classes = useStyles();

  const { months, years } = useMemo(
    () => getBirthAvailableMonthsAndYears(child.month, child.year, allow9MonthsOutFromNow),
    [child.year, child.month, allow9MonthsOutFromNow]
  );

  const monthsOptions = useMemo(() => {
    return months.map((month) => {
      return (
        <MenuItem key={month.value} value={month.value}>
          {month.name}
        </MenuItem>
      );
    });
  }, [months]);

  const yearsOptions = useMemo(() => {
    return years.map((year: string) => {
      return (
        <MenuItem key={year} value={year}>
          {year}
        </MenuItem>
      );
    });
  }, [years]);

  return (
    <Grid
      // eslint-disable-next-line react/no-array-index-key
      item
      xs={12}
      className={`${
        i === 0
          ? `${classes.childContainer}`
          : `${classes.childContainer} ${classes.childContainerNth}`
      }`}>
      <Typography variant="h3">Child {i + 1}</Typography>
      {i !== 0 && (
        <Button className={classes.removeChildButton} onClick={onRemoveChild}>
          Remove Child
        </Button>
      )}
      <Grid item xs={12} className={classes.optionsContainer}>
        <Grid item xs={6} className={`${classes.selectContainer} ${classes.monthSelect}`}>
          <InputLabel id={`month-select-label-${i}`}>Birth month</InputLabel>
          <Select
            className={classes.fullWidth}
            labelId={`month-select-label-${i}`}
            id={`month-select-child-${i}`}
            name={`children[${i}].month`}
            onChange={onChange}
            value={child.month}
            error={Boolean(error?.month)}>
            {monthsOptions}
          </Select>
          {error?.month && (
            <FormHelperText className={classes.error}>Select their birth month</FormHelperText>
          )}
        </Grid>
        <Grid item xs={6} className={`${classes.selectContainer} ${classes.yearSelect}`}>
          <InputLabel id={`year-select-label-${i}`}>Birth year</InputLabel>
          <Select
            className={classes.fullWidth}
            labelId={`year-select-label-${i}`}
            id={`year-select-child-${i}`}
            name={`children[${i}].year`}
            onChange={onChange}
            value={child.year}
            error={Boolean(error?.year)}>
            {yearsOptions}
          </Select>
          {error?.year && (
            <FormHelperText className={classes.error}>Select their birth year</FormHelperText>
          )}
        </Grid>
      </Grid>
    </Grid>
  );
}

ChildBirthdayFields.defaultProps = {
  error: undefined,
  allow9MonthsOutFromNow: undefined,
};

export type { IChildBirthdayFieldsProps, TErrorProps };
export default ChildBirthdayFields;
