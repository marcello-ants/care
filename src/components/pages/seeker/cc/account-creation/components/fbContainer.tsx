import { Box, makeStyles, Grid, Button } from '@material-ui/core';
import { useFacebookApiKey } from '@/components/FacebookApiKeyContext';

// @ts-ignore
import FacebookLogin from 'react-facebook-login/dist/facebook-login-render-props';
import { Icon48SocialMediaFacebook } from '@care/react-icons';

export type facebookResponseProps = {
  accessToken: string;
  // eslint-disable-next-line
  data_access_expiration_time?: number;
  email: string;
  expiresIn?: number;
  graphDomain?: string;
  id?: string;
  name: string;
  signedRequest?: string;
  userID?: string;
};

export interface SplitFormValues {
  email: string;
}

export interface SplitFormWithFbData extends SplitFormValues {
  firstName?: string;
  lastName?: string;
  accessToken?: string;
}

const mobileWidth = 360;
const fbColor = '#1877F2';
const fbIconSize = '35px';
const useStyles = (fbEmphasizedDesign: boolean) =>
  makeStyles((theme) => ({
    fbButton: {
      paddingTop: theme.spacing(5),
      borderColor: fbEmphasizedDesign ? fbColor : theme.palette.default?.dark,
      background: fbEmphasizedDesign ? fbColor : theme.palette.care?.white,
      color: fbEmphasizedDesign ? theme.palette.care?.white : theme.palette.default?.dark,
      fontSize: fbEmphasizedDesign ? '21px' : '18px',
      '&:hover': {
        background: fbEmphasizedDesign ? fbColor : theme.palette.care?.white,
      },
    },
    fbButtonContent: {
      display: 'grid',
      gridTemplateColumns: 'repeat(3, minmax(0, 1fr))',
      [theme.breakpoints.down(mobileWidth)]: {
        display: 'flex',
        alignItems: 'center',
      },
    },
    fbIconContainer: {
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'center',
      width: fbIconSize,
      '& svg': {
        width: fbIconSize,
        height: fbIconSize,
      },
    },
    fbIcon: {
      background: '#1a77f2',
      borderRadius: '19px',
    },
    fbButtonText: {
      justifySelf: 'center',
      marginLeft: '10px',
      marginRight: '5px',
      fontColor: theme.palette.care?.white,
    },
  }));

type GetFbContainerProps = {
  handleFacebookLogin: (response: facebookResponseProps) => Promise<void>;
  fbButtonOnTop: boolean;
  fbButtonOnBottom: boolean;
};

const getFbContainer = ({
  handleFacebookLogin,
  fbButtonOnTop,
  fbButtonOnBottom,
}: GetFbContainerProps) => {
  const fbEmphasizedDesign = fbButtonOnBottom || fbButtonOnTop;
  const classes = useStyles(fbEmphasizedDesign)();
  const { apiKey } = useFacebookApiKey();

  return (
    <Grid item xs={12}>
      <Box pt={fbButtonOnTop ? 0 : 2}>
        <FacebookLogin
          appId={apiKey}
          autoLoad={false}
          fields="name,email"
          callback={handleFacebookLogin}
          render={(renderProps: any) => (
            <Button
              onClick={renderProps.onClick}
              className={classes.fbButton}
              variant="outlined"
              size="large"
              fullWidth>
              <div className={classes.fbButtonContent}>
                <div className={classes.fbIconContainer}>
                  <Icon48SocialMediaFacebook size="38px" color="white" className={classes.fbIcon} />
                </div>
                <p className={classes.fbButtonText}>Continue with Facebook</p>
              </div>
            </Button>
          )}
        />
      </Box>
    </Grid>
  );
};

export default getFbContainer;
