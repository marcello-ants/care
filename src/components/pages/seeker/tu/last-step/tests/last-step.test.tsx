import { NextRouter, useRouter } from 'next/router';
import { render, waitFor, screen } from '@testing-library/react';
import { MockedProvider, MockedResponse } from '@apollo/client/testing';
import { initialAppState } from '@/state';
import { AppStateProvider, useAppDispatch } from '@/components/AppState';
import LastStep from '@/components/pages/seeker/tu/last-step/last-step';
import { AppState } from '@/types/app';
import logger from '@/lib/clientLogger';
import userEvent from '@testing-library/user-event';
import { TUTORING_RECURRING_JOB_CREATE } from '@/components/request/GQL';
import { generateTutoringJobCreateInput } from '@/utilities/gqlPayloadHelper';
import { ServerError } from '@apollo/client';
import { TealiumData, TealiumUtagService } from '@/utilities/utagHelper';

jest.mock('next/router', () => ({
  __esModule: true,
  useRouter: jest.fn(),
}));

jest.mock('next/config', () => () => {
  return {
    publicRuntimeConfig: {
      OIDC_STS_AUTHORITY: 'OIDC_STS_AUTHORITY',
      OIDC_CLIENT_ID: 'OIDC_CLIENT_ID',
      OIDC_RESPONSE_TYPE: 'OIDC_RESPONSE_TYPE',
      OIDC_CLIENT_SCOPE: 'OIDC_CLIENT_SCOPE',
      OIDC_CALLBACK_PATH: 'OIDC_CALLBACK_PATH',
      OIDC_LOGOUT_URL: 'OIDC_LOGOUT_URL',
      CZEN_GENERAL: '',
    },
  };
});

let mockRouter: Pick<NextRouter, 'push' | 'pathname'>;

jest.mock('@/utilities/utagHelper', () => ({
  __esModule: true,
  TealiumUtagService: {
    view: jest.fn(),
  },
}));
const tealiumMock = TealiumUtagService.view as jest.Mock;

jest.mock('@/components/AppState', () => ({
  ...(jest.requireActual('@/components/AppState') as object),
  useAppDispatch: jest.fn(),
}));
let mockAppDispatch: ReturnType<typeof useAppDispatch>;

const reducer = {
  prefix: 'tu',
  type: 'job_reducer',
};

const initialState: AppState = {
  ...initialAppState,
  seeker: {
    ...initialAppState.seeker,
    zipcode: '02452',
  },
  seekerTU: {
    ...initialAppState.seekerTU,
  },
};

const postAJobInput = generateTutoringJobCreateInput(initialState.seekerTU, initialState.seeker);
const postAJobMock: MockedResponse = {
  request: {
    query: TUTORING_RECURRING_JOB_CREATE,
    variables: {
      input: postAJobInput,
    },
  },
  result: {
    data: {
      job: { id: 123 },
    },
  },
};

const authError: ServerError = {
  statusCode: 401,
} as any;
const postAJobAuthFailureMock: MockedResponse = {
  ...postAJobMock,
  error: authError,
};

function renderPage(state: AppState, mocks: MockedResponse[] = []) {
  return render(
    <MockedProvider mocks={mocks} addTypename={false}>
      <AppStateProvider initialStateOverride={state}>
        <LastStep />
      </AppStateProvider>
    </MockedProvider>
  );
}

describe('Seeker - Tutoring - last step', () => {
  const realLoggerWarn = logger.warn;
  const realLoggerError = logger.error;

  const loggerWarnMock = jest.fn();
  const loggerErrorMock = jest.fn();

  beforeAll(() => {
    logger.warn = loggerWarnMock;
    logger.error = loggerErrorMock;
  });

  afterAll(() => {
    logger.warn = realLoggerWarn;
    logger.error = realLoggerError;
  });

  beforeEach(() => {
    jest.clearAllMocks();
    mockAppDispatch = jest.fn();
    (useAppDispatch as jest.Mock).mockReturnValue(mockAppDispatch);

    mockRouter = {
      push: jest.fn(),
      pathname: '/seeker/tu/last-step',
    };
    (useRouter as jest.Mock).mockReturnValue(mockRouter);

    loggerWarnMock.mockReset();
    loggerErrorMock.mockReset();
  });

  it('matches snapshot', async () => {
    const { asFragment } = renderPage(initialState);
    const nextButton = screen.getByRole('button', { name: 'Submit' });
    await waitFor(() => expect(nextButton).toBeEnabled());
    expect(asFragment()).toMatchSnapshot();
  });

  it('Filter our invalid chars from description', async () => {
    renderPage(initialState);
    const description = screen.getByRole('textbox');
    userEvent.paste(description, 'Nice!  Ὣ♂');
    expect(mockAppDispatch).toHaveBeenCalledWith({
      type: 'job_setJobDescription',
      description: 'Nice!  ',
      reducer,
    });
  });

  it('should redirect the user to pre-rate card page by clicking on Submit button when job was posted', async () => {
    // @ts-ignore
    delete window.location;
    /* eslint-disable no-restricted-globals */
    (window.location as Pick<typeof window.location, 'assign'>) = {
      assign: jest.fn(),
    };

    renderPage(initialState, [postAJobMock]);
    const submitButton = screen.getByRole('button', { name: 'Submit' });
    await waitFor(() => expect(submitButton).toBeEnabled());

    userEvent.click(submitButton);

    await waitFor(() =>
      expect(mockAppDispatch).toHaveBeenCalledWith({
        type: 'job_setJobSubmissionInfo',
        submissionAttempted: true,
        attempts: 1,
        reducer,
      })
    );

    await waitFor(() =>
      expect(mockAppDispatch).toHaveBeenCalledWith({
        type: 'job_setJobPostSuccessful',
        jobSuccessful: true,
        reducer,
      })
    );

    await waitFor(
      () =>
        expect(window.location.assign).toHaveBeenCalledWith(
          '/vis/auth/login?forwardUrl=%2Fapp%2Fratecard%2Ftutoring%2Fpre-rate-card%3FenrollFlow%3Dtrue'
        ),
      { timeout: 1500 }
    );
  });

  it('should log a warning after the user session after Submit click results in an auth error', async () => {
    renderPage(initialState, [postAJobAuthFailureMock]);
    const submitButton = screen.getByRole('button', { name: 'Submit' });
    await waitFor(() => expect(submitButton).toBeEnabled());

    userEvent.click(submitButton);

    await waitFor(() =>
      expect(mockAppDispatch).toHaveBeenCalledWith({
        type: 'job_setJobSubmissionInfo',
        submissionAttempted: true,
        attempts: 1,
        reducer,
      })
    );

    await waitFor(() => expect(loggerErrorMock).not.toHaveBeenCalled());
    await waitFor(() => expect(loggerWarnMock).toHaveBeenCalledTimes(1));
    await waitFor(() =>
      expect(loggerWarnMock).toHaveBeenCalledWith({
        event: 'postAJobErrorFrom401',
        graphQLError: 'undefined',
      })
    );
  });

  it('should execute PAJ request on page load if an attempt was previously made by the user', async () => {
    const state = {
      ...initialState,
      seekerTU: {
        ...initialState.seekerTU,
        jobPost: {
          ...initialState.seekerTU.jobPost,
          submissionInfo: {
            ...initialState.seekerTU.jobPost.submissionInfo,
            submissionAttempted: true,
            attempts: 1,
          },
        },
      },
    };

    renderPage(state, [postAJobMock]);

    await waitFor(() =>
      expect(mockAppDispatch).toHaveBeenCalledWith({
        type: 'job_setJobSubmissionInfo',
        submissionAttempted: false,
        attempts: 0,
        reducer,
      })
    );

    await waitFor(() =>
      expect(mockAppDispatch).toHaveBeenCalledWith({
        type: 'job_setJobPostSuccessful',
        jobSuccessful: true,
        reducer,
      })
    );

    await waitFor(
      () =>
        expect(window.location.assign).toHaveBeenCalledWith(
          '/vis/auth/login?forwardUrl=%2Fapp%2Fratecard%2Ftutoring%2Fpre-rate-card%3FenrollFlow%3Dtrue'
        ),
      { timeout: 1500 }
    );
  });

  it('should execute PAJ request on page load and log a warning if another auth error is received', async () => {
    const state = {
      ...initialState,
      seekerTU: {
        ...initialState.seekerTU,
        jobPost: {
          ...initialState.seekerTU.jobPost,
          submissionInfo: {
            ...initialState.seekerTU.jobPost.submissionInfo,
            submissionAttempted: true,
            attempts: 1,
          },
        },
      },
    };

    renderPage(state, [postAJobAuthFailureMock]);

    await waitFor(() =>
      expect(mockAppDispatch).toHaveBeenCalledWith({
        type: 'job_setJobSubmissionInfo',
        submissionAttempted: true,
        attempts: 2,
        reducer,
      })
    );

    await waitFor(() => expect(loggerErrorMock).not.toHaveBeenCalled());
    await waitFor(() => expect(loggerWarnMock).toHaveBeenCalledTimes(1));
    await waitFor(() =>
      expect(loggerWarnMock).toHaveBeenCalledWith({
        event: 'postAJobErrorFrom401',
        graphQLError: 'undefined',
      })
    );
  });

  it('should execute PAJ request on page load aand log an error when user is at max number of retries and if another auth error is received', async () => {
    const state = {
      ...initialState,
      seekerTU: {
        ...initialState.seekerTU,
        jobPost: {
          ...initialState.seekerTU.jobPost,
          submissionInfo: {
            ...initialState.seekerTU.jobPost.submissionInfo,
            submissionAttempted: true,
            attempts: 3,
          },
        },
      },
    };

    renderPage(state, [postAJobAuthFailureMock]);

    await waitFor(() => expect(loggerWarnMock).not.toHaveBeenCalled());
    await waitFor(() =>
      expect(loggerErrorMock).toHaveBeenCalledWith({
        event: 'postAJobMaxAttemptsFrom401',
        graphQLError: 'undefined',
      })
    );
    await waitFor(
      () =>
        expect(window.location.assign).not.toHaveBeenCalledWith(
          '/vis/auth/login?forwardUrl=/app/ratecard/tutoring/pre-rate-card?enrollFlow=true'
        ),
      { timeout: 1500 }
    );
  });

  it('sends the right data for Tealium event', async () => {
    const tealiumData: TealiumData = {
      sessionId: undefined,
      slots: ['/us-subscription/action/seeker/paj/tu'],
    };

    renderPage(initialState, [postAJobMock]);
    userEvent.click(screen.getByText('Submit'));

    await waitFor(
      async () => {
        expect(tealiumMock).toHaveBeenCalledWith(tealiumData);
      },
      { timeout: 1200 }
    );
  });

  it('should not allow user to post the same job', async () => {
    const state = {
      ...initialState,
      seekerTU: {
        ...initialState.seekerTU,
        jobPost: {
          ...initialState.seekerTU.jobPost,
          submissionInfo: {
            ...initialState.seekerTU.jobPost.submissionInfo,
            jobPostSuccessful: true,
          },
        },
      },
    };

    renderPage(state);
    const nextButton = screen.getByRole('button', { name: 'Continue' });
    await waitFor(() => expect(nextButton).toBeEnabled());
  });

  it('should redirect user to prerate-card without submitting job because of MAX_JOB_POST_ATTEMPTS', async () => {
    const state = {
      ...initialState,
      seekerTU: {
        ...initialState.seekerTU,
        jobPost: {
          ...initialState.seekerTU.jobPost,
          submissionInfo: {
            ...initialState.seekerTU.jobPost.submissionInfo,
            attempts: 3,
          },
        },
      },
    };

    renderPage(state);
    const nextButton = screen.getByRole('button', { name: 'Continue without submitting' });
    await waitFor(() => expect(nextButton).toBeEnabled());
    userEvent.click(nextButton);
    expect(window.location.assign).toHaveBeenCalledWith(
      '/vis/auth/login?forwardUrl=%2Fapp%2Fratecard%2Ftutoring%2Fpre-rate-card%3FenrollFlow%3Dtrue'
    );
  });
});
