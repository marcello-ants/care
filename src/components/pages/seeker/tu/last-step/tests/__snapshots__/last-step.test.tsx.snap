// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Seeker - Tutoring - last step matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root makeStyles-lastStepWrapper-3 MuiGrid-container"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <h2
        class="MuiTypography-root makeStyles-sectionOneTitle-2 MuiTypography-h2 MuiTypography-colorTextPrimary"
      >
        What else should tutors know?
      </h2>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        class="MuiFormControl-root MuiTextField-root makeStyles-about-4"
      >
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-multiline MuiInput-multiline"
        >
          <textarea
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline MuiInput-inputMultiline"
            maxlength="850"
            name="description"
            placeholder="Tell tutors a little more about any specific needs. (Optional)"
            rows="12"
          />
        </div>
      </div>
    </div>
    <div
      class="makeStyles-tipsContainer-22"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root makeStyles-root-39"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <defs>
          <clippath
            id="icon-24-info-tips-svg-a"
          >
            <path
              d="M4.8 2h14.4v20.01H4.8z"
              fill="none"
            />
          </clippath>
        </defs>
        <g
          clip-path="url(#icon-24-info-tips-svg-a)"
          fill="#182c42"
        >
          <path
            d="M15.1 22H8.8a.7.7 0 01-.49-.2.71.71 0 01.49-1.2h6.3a.71.71 0 01.5.21.7.7 0 01-.5 1.19zm.1-3.3H8.8a.59.59 0 01-.38-.14.65.65 0 01-.22-.36l-1-3.6v-.1a7.39 7.39 0 01-1.71-2.28 7.19 7.19 0 01-.68-2.77 7 7 0 01.42-2.76A7.18 7.18 0 016.7 4.3a7.47 7.47 0 012.42-1.7 7.19 7.19 0 019.55 3.84 7.1 7.1 0 01.53 2.76 8.69 8.69 0 01-2.2 5.3l-1.2 3.7a.63.63 0 01-.21.36.62.62 0 01-.39.14zm-4-1.3h3.4v-.2l1-3.2c0-.1.1-.2.2-.3a7.14 7.14 0 002-4.6 5.84 5.84 0 00-1.71-4.19A5.91 5.91 0 0011.9 3.2a5.74 5.74 0 00-2.27.44 5.85 5.85 0 00-1.92 1.27 6 6 0 00-1.28 1.93A5.82 5.82 0 006 9.1a6 6 0 002.1 4.5.65.65 0 01.15.18.59.59 0 01.05.22l.9 3.2v.2z"
          />
          <path
            d="M11.85 11.5a2.25 2.25 0 112.25-2.25 2.25 2.25 0 01-2.25 2.25zm0-3.3a1.05 1.05 0 101.05 1.05 1 1 0 00-1.05-1.05z"
          />
        </g>
        <path
          d="M11.8 10.7a.6.6 0 01.6.6v6.5a.6.6 0 01-.6.6.6.6 0 01-.6-.6v-6.5a.6.6 0 01.6-.6z"
          fill="#182c42"
        />
        <path
          d="M11.8 18.5a.71.71 0 01-.7-.7v-6.5a.7.7 0 011.4 0v6.5a.7.7 0 01-.7.7zm0-7.7a.5.5 0 00-.5.5v6.5a.51.51 0 00.5.5.5.5 0 00.5-.5v-6.5a.5.5 0 00-.5-.5z"
          fill="#182c42"
        />
      </svg>
      <p
        class="MuiTypography-root makeStyles-tipsButton-21 MuiTypography-body1"
      >
        Need some tips?
      </p>
    </div>
    <div
      class="MuiGrid-root makeStyles-nextButtonContainer-1 MuiGrid-item MuiGrid-grid-xs-12"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge MuiButton-fullWidth"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Submit
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
